lib/shanty/plugin.rb

```diff
@@ -33,6 +45,14 @@ def self.with_graph(&block)
       (@with_graph_callbacks ||= []) << block
     end
 
+    def self.name
+      to_s.split('::').last.downcase.gsub('plugin', '').to_sym
+    end
+
+    def name
+      self.class.name
```

Comment:
This is a bit rank - would just delete this and anybody who wants to get names can just call `.class.name` whne they want to.